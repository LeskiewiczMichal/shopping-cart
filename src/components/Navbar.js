import '../styles/navbar.css';
import { Link } from 'react-router-dom';
import shoppingCart from '../icons/shopping-cart.png';

const Navbar = (props) => {
  return (
    <nav className="navbar--wrapper">
      <div className="navbar--links-wrapper">
        <Link className="navbar--link" to="/">
          Home
        </Link>
        <Link className="navbar--link" to="shop">
          Shop
        </Link>
      </div>
      <h1>Shopper</h1>
      <Link className="navbar--link" to="shop" data-testid="secondShopLink">
        <div className="navbar--cart">
          <span>{props.cart.length}</span>
          <img
            className="navbar--icon"
            src={shoppingCart}
            alt="shopping cart"
          />
        </div>
      </Link>
    </nav>
  );
};

export default Navbar;
