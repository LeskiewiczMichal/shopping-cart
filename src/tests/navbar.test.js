import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';
import { MemoryRouter } from 'react-router-dom';

const cart = [];

test('Renders links', () => {
  render(<Navbar cart={cart} />, { wrapper: MemoryRouter });

  const homePageLink = screen.getByRole('link', { name: 'Home' });
  const firstShopPageLink = screen.getByRole('link', { name: 'Shop' });
  const secondShopPageLink = screen.getByTestId('secondShopLink');

  expect(homePageLink).toBeInTheDocument();
  expect(firstShopPageLink).toBeInTheDocument();
  expect(secondShopPageLink).toBeInTheDocument();
});

test('Renders page name', () => {
  render(<Navbar cart={cart} />, { wrapper: MemoryRouter });

  const heading = screen.getByRole('heading');
  expect(heading).toBeInTheDocument();
});

test('Renders cart length', () => {
  render(<Navbar cart={cart} />, { wrapper: MemoryRouter });

  const number = screen.getByText('0');
  expect(number).toBeInTheDocument();
});

test('Renders image', () => {
  render(<Navbar cart={cart} />, { wrapper: MemoryRouter });

  const image = screen.getByRole('img');
  expect(image).toBeInTheDocument();
});

test('Snapshot', () => {
  const { container } = render(<Navbar cart={cart} />, {
    wrapper: MemoryRouter,
  });
  expect(container).toMatchSnapshot();
});
