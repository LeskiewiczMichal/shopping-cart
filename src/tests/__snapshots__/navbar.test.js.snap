// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <nav
    class="navbar--wrapper"
  >
    <div
      class="navbar--links-wrapper"
    >
      <a
        class="navbar--link"
        href="/"
      >
        Home
      </a>
      <a
        class="navbar--link"
        href="/shop"
      >
        Shop
      </a>
    </div>
    <h1>
      Shopper
    </h1>
    <a
      class="navbar--link"
      data-testid="secondShopLink"
      href="/shop"
    >
      <div
        class="navbar--cart"
      >
        <span>
          0
        </span>
        <img
          alt="shopping cart"
          class="navbar--icon"
          src="shopping-cart.png"
        />
      </div>
    </a>
  </nav>
</div>
`;
